### PR TITLE
Add Structure.save function

### LINKED ISSUE #106: add universal `save` for Parm object
I am playing with ParmEd in pytraj namespace

    >>> from pytraj import io
    >>> p_top = io.load_ParmEd("./Tc5b.top")
    >>> p_top
        <AmberParm 304 atoms; 20 residues; 310 bonds; parametrized>

     >>> # it's really nice if ParmEd has this
     >>> p_top.save("Tc5b.cif")
     >>> p_top.save("Tc5b.psf")
     >>> p_top.save("Tc5b.crazy_ext", fmt='psf') 
     >>> # if fmt is not None, ParmEd will set filetype of fmt

(I know there are methods like `write_cif` in ParmEd)

### PR DESCRIPTION
**Pull request by swails**
This allows you to write the desired format in a *single* place, with any
necessary conversions being done for you automatically. It also handles
Amber-style formats as well as possible, determining in most instances whether a
structure can be cast as a ChamberParm when AmberParm is not flexible enough.
File type is determined from filename extension *or* explicit direction.

This also fixes writing PSF files from Structure instances that were *not*
parsed from PSFs (since the `groups` list did not have a `nst2` attribute.

This fixes #106.

### CODE REVIEW

**Comment by swails** [comment]
Still needs tests.

**Comment by hainm** [comment]
+1

File: parmed/structure.py [hunk 1]
```diff
@@ -1293,6 +1292,104 @@ def split(self):
 
     #===================================================
 
+    def save(self, fname, kind=None, **kwargs):
```
Comment (hainm) [inline parmed/structure.py]: why not using 'format="pdb"' instead of using 'kind'?

Comment (swails) [inline parmed/structure.py]: Format is a better choice. Kind was just the first word that came to mind when I was coding it.

**Comment by swails** [comment]
Travis spuriously killed one of the build matrix.  Merging anyway.

**Comment by hainm** [comment]
it's really convenient to use parmed now.

think about reading gaussian output, .ac, ... files? I am working on modified nuc, there are many steps and programs involve to derive new pars. I wish I only need to remember "parmed". lol.

**Comment by swails** [comment]
I'd like to have parsers for everything, but Gaussian is below many others on the list currently.

**Comment by leeping** [comment]
Hi Hai,

I think that reading / writing quantum chemistry formats is challenging but doable.  The output depends on the type of job being run (energy, gradient, optimization, and properties being a few examples).  There is also the question of what information we should read from the file (e.g. SCF energy, MP2 correlation energy, Mulliken charges, intermediate geometries from optimization being a few examples).  When writing Gaussian input, one needs to specify the type of job being run and other parameters such as total memory. This is all interesting stuff to think about.

Thanks,

- Lee-Ping

**Comment by hainm** [comment]
thanks

Hai

**Pull request merged**